### PR TITLE
Remove parallel option for FlowGroup submission

### PR DESCRIPTION
**Pull request by b-butler**
## Description
Removes the option to run `FlowGroup`s in parallel this does not effect inter-group parallelization.

## Motivation and Context
This removes the potentially unexpected behavior of parallelizing both within and across groups when
the `parallel` flag is used #270. A more robust implementation that supports both kinds of
parallelism needs to be created for the future, but in order to release flow this should provide the
safe option (inter-group) parallelism.

## Types of Changes

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

Note that I don't think this should be on the changelog since we have not released `FlowGroup`s yet,
and the current change log does not mention parallelization.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by b-butler** [comment]
Merge after #279.

**Comment by b-butler** [comment]
@bdice @pepak13 this is ready for review.

File: flow/project.py [hunk 1]
```diff
@@ -2804,9 +2789,10 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
 
         # Gather all pending operations.
         with self._potentially_buffered():
+
```
Comment (bdice) [inline flow/project.py:2792]: Remove extra newline.
```suggestion
```

**Comment by bdice** [comment]
@b-butler ⚠️ This will require a corresponding PR to signac-docs to fix this section: https://docs.signac.io/en/latest/flow-group.html?highlight=flowgroup#running-groups-in-parallel

I don't think we need to update the changelog since we're altering an unreleased feature, right?

**Comment by b-butler** [comment]
@bdice that was my thoughts regarding the changelog. I just wanted to mention that this PR shouldn't have a changelog entry. I will check your PR assuming, I don't see anything merge both.

**Pull request merged**